### PR TITLE
changes only mode

### PR DESCRIPTION
**Pull request by Bonczidai**
Signed-off-by: Bonczidai Levente <levente.bonczidai@hpe.com>

### CODE REVIEW

File: cloudslang-content-verifier/pom.xml [hunk 1]
```diff
@@ -18,6 +18,12 @@
         </dependency>
 
         <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>cloudslang-entities</artifactId>
```
Comment (lucian-cm) [inline cloudslang-content-verifier/pom.xml]: <ignoredUsedUndeclaredDependency>io.cloudslang.lang:cloudslang-entities:jar:${project.version}</ignoredUsedUndeclaredDependency> should be removed from this module

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/SlangBuilder.java [hunk 2]
```diff
@@ -124,30 +140,56 @@ IRunTestResults runTests(Map<String, Executable> contentSlangModels,
...
 
         runTestsResults.addExceptions(compilationResult.getExceptions());
         addCoverageDataToRunTestsResults(contentSlangModels, testFlowModels, testCases, runTestsResults);
         return runTestsResults;
     }
 
-    IRunTestResults processRunTests(String projectPath, List<String> testSuites, BulkRunMode bulkRunMode, Map<String, CompilationArtifact> compiledFlows, Map<String, SlangTestCase> testCases) {
+    private BuildModeConfig createBuildModeConfig(SlangBuildMain.BuildMode buildMode, Set<String> changedFiles, Map<String, Executable> allTestedFlowModels) {
+        BuildModeConfig buildModeConfig;
+        switch (buildMode) {
```
Comment (lucian-cm) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/SlangBuilder.java]: if else if is a bit shorter in lines of code then switch, no need to change this.

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/commands/ApplicationArgs.java [hunk 3]
```diff
@@ -22,6 +21,8 @@
 
 public class ApplicationArgs {
 
+    private static final String DYNAMIC_PARAMETER_CHANGES_ONLY = "changes_only";
```
Comment (lucian-cm) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/commands/ApplicationArgs.java]: maybe it is worth showing this as a regular parameter to get more visibility. WDYT?

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java [hunk 4]
```diff
@@ -242,7 +253,8 @@ private void printTestForActualRunSummary(TestCaseRunMode runMode, Map<String, S
                 continue;
             }
 
-            if (isTestCaseInActiveSuite(testCase, testSuites)) {
+            if (isTestCaseInActiveSuite(testCase, testSuites) &&
+                    enabledByBuildMode(buildModeConfig.getBuildMode(), testCase, buildModeConfig.getChangedFiles(), buildModeConfig.getAllTestedFlowModels())) {
```
Comment (lucian-cm) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java]: Maybe name this `isEnabledByBuildMode`

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java [hunk 5]
```diff
@@ -252,6 +264,32 @@ private void printTestForActualRunSummary(TestCaseRunMode runMode, Map<String, S
         return resultMap;
     }
 
+    private boolean enabledByBuildMode(
+            SlangBuildMain.BuildMode buildMode,
+            SlangTestCase slangTestCase,
+            Set<String> changedExecutables,
+            Map<String, Executable> allTestedFlowModels) {
+        switch (buildMode) {
```
Comment (lucian-cm) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java]: I think this method can be replaced (as the `buildMode `is already validated somewhere in the begining of this flow) with `return (buildMode == BASIC) || ((buildMode == CHANGED) && isAffectedTestCase(slangTestCase, changedExecutables, allTestedFlowModels))` WDYT?

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java [hunk 6]
```diff
@@ -252,6 +264,32 @@ private void printTestForActualRunSummary(TestCaseRunMode runMode, Map<String, S
...
+        }
+    }
+
+    private boolean isAffectedTestCase(SlangTestCase slangTestCase, Set<String> changedExecutables, Map<String, Executable> allTestedFlowModels) {
+        String testFlowPath = slangTestCase.getTestFlowPath();
+        Executable testCaseReference = allTestedFlowModels.get(testFlowPath);
+        if (testCaseReference == null) {
+            throw new RuntimeException("Test case reference[" + testFlowPath + "] not found in compiled models.");
+        }
+        Set<String> testCaseDependencies = new HashSet<>(dependenciesHelper.fetchDependencies(testCaseReference, allTestedFlowModels));
```
Comment (lucian-cm) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java]: Maybe I am missing something, but I think from the github interface that this is shallow cloning the fetchDependencies set with another set, so please ignore if this is not the case.

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/DependenciesHelper.java [hunk 7]
```diff
@@ -50,6 +51,34 @@
...
+
+        switch (executable.getType()) {
+            case SlangTextualKeys.OPERATION_TYPE:
+                return Collections.emptySet();
+            case SlangTextualKeys.DECISION_TYPE:
+                return Collections.emptySet();
+            case SlangTextualKeys.FLOW_TYPE:
+                return processFlowForDependencies((Flow) executable, availableDependencies);
+            default:
+                throw new RuntimeException(MessageConstants.NOT_IMPLEMENTED_MESSAGE);
```
Comment (lucian-cm) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/DependenciesHelper.java]: Maybe you can use `org.apache.commons.lang3.NotImplementedException`. WDYT?

Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/DependenciesHelper.java]: 👍

File: cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java (hunk 5 above)
Comment (Bonczidai) [inline cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/tester/SlangTestRunner.java]: 👍

**Comment by Bonczidai** [comment]
thanks @lucian-cm for the review

**Pull request merged**